From d86af63d8142f95379b3954ad80f1682a2398135 Mon Sep 17 00:00:00 2001
From: "Yue, VincentX" <vincentx.yue@intel.com>
Date: Fri, 12 Jan 2018 17:43:21 +0800
Subject: [PATCH 1/3] Fix BT paired icon is not displayed on the New User's
 status bar

When switch to new user, systemUI is owner process, unable to
get the bluetooth connectition device update information throung broadcast.
so, send the connection information to all user like intent ACTION_CONNECTION_STATE_CHANGED.

Change-Id: I70d56abba97bd22d56a7dec4484590723eb02da1
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-54807
Signed-off-by: Yue, VincentX <vincentx.yue@intel.com>
---
 src/com/android/bluetooth/hfpclient/HeadsetClientStateMachine.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/com/android/bluetooth/hfpclient/HeadsetClientStateMachine.java b/src/com/android/bluetooth/hfpclient/HeadsetClientStateMachine.java
index 180d91c..0785c92 100755
--- a/src/com/android/bluetooth/hfpclient/HeadsetClientStateMachine.java
+++ b/src/com/android/bluetooth/hfpclient/HeadsetClientStateMachine.java
@@ -47,6 +47,7 @@ import android.os.Message;
 import android.os.Looper;
 import android.os.ParcelUuid;
 import android.os.SystemClock;
+import android.os.UserHandle;
 import android.util.Log;
 import android.util.Pair;
 import android.telecom.TelecomManager;
@@ -1709,7 +1710,7 @@ public class HeadsetClientStateMachine extends StateMachine {
                 intent.putExtra(BluetoothHeadsetClient.EXTRA_AG_FEATURE_MERGE_AND_DETACH, true);
             }
         }
-        mService.sendBroadcast(intent, ProfileService.BLUETOOTH_PERM);
+        mService.sendBroadcastAsUser(intent, UserHandle.ALL, ProfileService.BLUETOOTH_PERM);
     }
 
     boolean isConnected() {
-- 
1.9.1

